salt/minion.py

```diff
@@ -2926,7 +2930,38 @@ def nodegroup_match(self, tgt, nodegroups):
         return False
 
 
-class ProxyMinion(MinionManager):
+class ProxyMinionManager(MinionManager):
+    '''
+    Create the multi-minion interface but for proxy minions
+    '''
+
+    def _spawn_minions(self):
+        '''
+        Spawn all the coroutines which will sign in to masters
+        NOTE MinionManager._spawn_minions should be examined if any changes are made to
```

Comment:
This is a red flag to me that we need to refactor this. Could you elaborate on what's different between this and the overriden method?